### PR TITLE
fix(leadspace): add super variation and tag group

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#5960 #6153 #6160

### Description

Adds tag group to leadspace , fixes height issues and heading text for each variation. 
Screen Shot 2021-05-24 at 8 34 56 AM

### Changelog

**New**

- slot for navigations elements above heading
- handleSlotChange if tag group or breadcrumbs in slot
- add super variation 

**Changed**

- style updates

### CODE REVIEW

**Comment by shixiedesign** [comment]
Hi Ari! Thank you! Looks good and only 2 small adjustments: 
1. Let's not allow Carbon tags as an option in Lead space. Carbon tag is not interactive at the moment and too small. 
image

2. Can we show only 2 tags in the group? Ideally we don't see too many tags in lead space out in the wild so let's demo the ideal usage. 

Thanks!

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 1]
```diff
@@ -94,25 +90,24 @@ $btn-min-width: 26;
...
       height: 100%;
     }
 
-    .#{$prefix}--leadspace--content__container--medium {
-      display: flex;
-      flex-direction: column;
-      justify-content: space-between;
-      height: 100%;
+    .#{$prefix}--leadspace__section {
+      height: 560px;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_leadspace.scss:110]:
```suggestion
      height: rem(560px);
```

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 2]
```diff
@@ -294,20 +256,19 @@ $btn-min-width: 26;
       }
 
       &.#{$prefix}--leadspace__section {
-        height: 100%;
+        height: 560px;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_leadspace.scss:259]:
```suggestion
        height: rem(560px);
```

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 3]
```diff
@@ -372,38 +323,29 @@ $btn-min-width: 26;
           padding: 0 0 carbon--mini-units(20) 0;
         }
 
+        .#{$prefix}--leadspace--content__container--super,
+        .#{$prefix}--leadspace--content__container--medium,
         .#{$prefix}--leadspace--content__container {
           margin: 0 auto;
           padding-left: 1.5rem;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_leadspace.scss:330]: Should use spacing token here.

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 4]
```diff
@@ -488,19 +430,40 @@ $btn-min-width: 26;
     }
   }
 
+  :host(#{$dds-prefix}-leadspace) ::slotted([slot='extra-elements']) {
+    z-index: 1;
+    padding-left: 1rem;
+    padding-bottom: 0.5rem;
+    max-width: 640px;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_leadspace.scss:437]: Use spacing tokens for padding and `rem()` for width.

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 5]
```diff
@@ -488,19 +430,40 @@ $btn-min-width: 26;
...
+  }
+
   :host(#{$dds-prefix}-leadspace-heading),
   .#{$prefix}--leadspace__title {
     padding-top: 0;
     margin-bottom: $spacing-09;
     padding-left: 1rem;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_leadspace.scss:444]: Use spacing token here.

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 6]
```diff
@@ -488,19 +430,40 @@ $btn-min-width: 26;
...
-  :host(#{$dds-prefix}-leadspace-heading)[size='medium'] {
-    margin-bottom: $spacing-06;
-    @include carbon--type-style(display-01, true);
+  :host(#{$dds-prefix}-leadspace)[size='medium'] {
+    ::slotted(#{$dds-prefix}-leadspace-heading) {
+      margin-bottom: $spacing-06;
+      @include carbon--type-style(expressive-heading-05, true);
+    }
+    .#{$prefix}--leadspace__section {
+      height: 480px;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_leadspace.scss:456]:
```suggestion
      height: rem(480px);
```

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 7]
```diff
@@ -488,19 +430,40 @@ $btn-min-width: 26;
...
+    }
+  }
+
+  :host(#{$dds-prefix}-leadspace)[size='super'] {
+    ::slotted(#{$dds-prefix}-leadspace-heading) {
+      @include carbon--type-style(display-01, true);
+    }
+    .#{$prefix}--leadspace__section {
+      height: 640px;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_leadspace.scss:465]:
```suggestion
      height: rem(640px);
```

**Review by kennylam — COMMENTED** [review]
Thanks @ariellalgilmore! Just a few small things.

File: packages/styles/scss/components/leadspace/_leadspace.scss (hunk 5 above)
Comment (ariellalgilmore) [reply packages/styles/scss/components/leadspace/_leadspace.scss]: update all the tokens and added rem! thanks Kenny!

**Review by kennylam — APPROVED** [review]
Looks good thanks @ariellalgilmore!

**Comment by photodow** [comment]
Tag link knob doesn't appear to be working on the centered lead space stories. I also just want to clarify this comment here "slot for extra elements above heading" because it sounds like anything can go up there, and we definitely don't want anything to go in this slot. Only tag links.

**Comment by ariellalgilmore** [comment]
> Tag link knob doesn't appear to be working on the centered lead space stories. I also just want to clarify this comment here "slot for extra elements above heading" because it sounds like anything can go up there, and we definitely don't want anything to go in this slot. Only tag links.

updated!! only tag group and breadcrumbs can go above heading

**Review by shixiedesign — APPROVED** [review]
Looks good to me!!! ⚡

**Pull request merged**